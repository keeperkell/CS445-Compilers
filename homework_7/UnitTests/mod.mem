Func: input returns type int [mem: Global loc: 0 size: -2] [line: -1]
Sibling: 1  Func: output returns type void [mem: Global loc: 0 size: -3] [line: -1]
.   Child: 0  Parm: *dummy* of type int [mem: Parameter loc: -2 size: 1] [line: -1]
Sibling: 2  Func: inputb returns type bool [mem: Global loc: 0 size: -2] [line: -1]
Sibling: 3  Func: outputb returns type void [mem: Global loc: 0 size: -3] [line: -1]
.   Child: 0  Parm: *dummy* of type bool [mem: Parameter loc: -2 size: 1] [line: -1]
Sibling: 4  Func: inputc returns type char [mem: Global loc: 0 size: -2] [line: -1]
Sibling: 5  Func: outputc returns type void [mem: Global loc: 0 size: -3] [line: -1]
.   Child: 0  Parm: *dummy* of type char [mem: Parameter loc: -2 size: 1] [line: -1]
Sibling: 6  Func: outnl returns type void [mem: Global loc: 0 size: -2] [line: -1]
Sibling: 7  Func: main returns type void [mem: Global loc: 0 size: -2] [line: 1]
.   Child: 1  Compound [mem: None loc: 0 size: -2] [line: 2]
.   .   Child: 1  Call: output of type void [line: 3]
.   .   .   Child: 0  Op: % of type int [line: 3]
.   .   .   .   Child: 0  Const 10 of type int [line: 3]
.   .   .   .   Child: 1  Const 1 of type int [line: 3]
.   .   Sibling: 1  Call: output of type void [line: 4]
.   .   .   Child: 0  Op: % of type int [line: 4]
.   .   .   .   Child: 0  Const 10 of type int [line: 4]
.   .   .   .   Child: 1  Const 2 of type int [line: 4]
.   .   Sibling: 2  Call: output of type void [line: 5]
.   .   .   Child: 0  Op: % of type int [line: 5]
.   .   .   .   Child: 0  Const 10 of type int [line: 5]
.   .   .   .   Child: 1  Const 3 of type int [line: 5]
.   .   Sibling: 3  Call: output of type void [line: 6]
.   .   .   Child: 0  Op: % of type int [line: 6]
.   .   .   .   Child: 0  Const 10 of type int [line: 6]
.   .   .   .   Child: 1  Const 4 of type int [line: 6]
.   .   Sibling: 4  Call: output of type void [line: 7]
.   .   .   Child: 0  Op: % of type int [line: 7]
.   .   .   .   Child: 0  Const 10 of type int [line: 7]
.   .   .   .   Child: 1  Const 5 of type int [line: 7]
.   .   Sibling: 5  Call: outnl of type void [line: 8]
.   .   Sibling: 6  Call: output of type void [line: 9]
.   .   .   Child: 0  Op: % of type int [line: 9]
.   .   .   .   Child: 0  Const 10 of type int [line: 9]
.   .   .   .   Child: 1  Const 10 of type int [line: 9]
.   .   Sibling: 7  Call: output of type void [line: 10]
.   .   .   Child: 0  Op: % of type int [line: 10]
.   .   .   .   Child: 0  Const 10 of type int [line: 10]
.   .   .   .   Child: 1  Const 11 of type int [line: 10]
.   .   Sibling: 8  Call: outnl of type void [line: 11]
.   .   Sibling: 9  Call: output of type void [line: 13]
.   .   .   Child: 0  Op: % of type int [line: 13]
.   .   .   .   Child: 0  Const 0 of type int [line: 13]
.   .   .   .   Child: 1  Const 1 of type int [line: 13]
.   .   Sibling: 10  Call: outnl of type void [line: 14]
.   .   Sibling: 11  Call: output of type void [line: 16]
.   .   .   Child: 0  Op: % of type int [line: 16]
.   .   .   .   Child: 0  Op: chsign of type int [line: 16]
.   .   .   .   .   Child: 0  Const 10 of type int [line: 16]
.   .   .   .   Child: 1  Const 1 of type int [line: 16]
.   .   Sibling: 12  Call: output of type void [line: 17]
.   .   .   Child: 0  Op: % of type int [line: 17]
.   .   .   .   Child: 0  Op: chsign of type int [line: 17]
.   .   .   .   .   Child: 0  Const 10 of type int [line: 17]
.   .   .   .   Child: 1  Const 2 of type int [line: 17]
.   .   Sibling: 13  Call: output of type void [line: 18]
.   .   .   Child: 0  Op: % of type int [line: 18]
.   .   .   .   Child: 0  Op: chsign of type int [line: 18]
.   .   .   .   .   Child: 0  Const 10 of type int [line: 18]
.   .   .   .   Child: 1  Const 3 of type int [line: 18]
.   .   Sibling: 14  Call: output of type void [line: 19]
.   .   .   Child: 0  Op: % of type int [line: 19]
.   .   .   .   Child: 0  Op: chsign of type int [line: 19]
.   .   .   .   .   Child: 0  Const 10 of type int [line: 19]
.   .   .   .   Child: 1  Const 4 of type int [line: 19]
.   .   Sibling: 15  Call: output of type void [line: 20]
.   .   .   Child: 0  Op: % of type int [line: 20]
.   .   .   .   Child: 0  Op: chsign of type int [line: 20]
.   .   .   .   .   Child: 0  Const 10 of type int [line: 20]
.   .   .   .   Child: 1  Const 5 of type int [line: 20]
.   .   Sibling: 16  Call: outnl of type void [line: 21]
.   .   Sibling: 17  Call: output of type void [line: 22]
.   .   .   Child: 0  Op: % of type int [line: 22]
.   .   .   .   Child: 0  Op: chsign of type int [line: 22]
.   .   .   .   .   Child: 0  Const 10 of type int [line: 22]
.   .   .   .   Child: 1  Const 10 of type int [line: 22]
.   .   Sibling: 18  Call: output of type void [line: 23]
.   .   .   Child: 0  Op: % of type int [line: 23]
.   .   .   .   Child: 0  Op: chsign of type int [line: 23]
.   .   .   .   .   Child: 0  Const 10 of type int [line: 23]
.   .   .   .   Child: 1  Const 11 of type int [line: 23]
.   .   Sibling: 19  Call: outnl of type void [line: 24]
.   .   Sibling: 20  Call: output of type void [line: 26]
.   .   .   Child: 0  Op: % of type int [line: 26]
.   .   .   .   Child: 0  Const 331 of type int [line: 26]
.   .   .   .   Child: 1  Const 31 of type int [line: 26]
.   .   Sibling: 21  Call: output of type void [line: 27]
.   .   .   Child: 0  Op: % of type int [line: 27]
.   .   .   .   Child: 0  Const 331 of type int [line: 27]
.   .   .   .   Child: 1  Op: chsign of type int [line: 27]
.   .   .   .   .   Child: 0  Const 31 of type int [line: 27]
.   .   Sibling: 22  Call: output of type void [line: 28]
.   .   .   Child: 0  Op: % of type int [line: 28]
.   .   .   .   Child: 0  Op: chsign of type int [line: 28]
.   .   .   .   .   Child: 0  Const 331 of type int [line: 28]
.   .   .   .   Child: 1  Const 31 of type int [line: 28]
.   .   Sibling: 23  Call: output of type void [line: 29]
.   .   .   Child: 0  Op: % of type int [line: 29]
.   .   .   .   Child: 0  Op: chsign of type int [line: 29]
.   .   .   .   .   Child: 0  Const 331 of type int [line: 29]
.   .   .   .   Child: 1  Op: chsign of type int [line: 29]
.   .   .   .   .   Child: 0  Const 31 of type int [line: 29]
Offset for end of global space: 0
Number of warnings: 0
Number of errors: 0
