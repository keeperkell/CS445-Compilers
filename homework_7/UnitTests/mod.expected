Number of warnings: 0
Number of errors: 0
------------------------------------
Loading file: UnitTests/mod.tm
0 0 1 2 0
0 10
0
0 0 2 2 0
0 1
21 21 10 10
Bye.
